src/components/fields/ArrayField.js

```diff
@@ -88,68 +94,157 @@ class ArrayField extends Component {
+    const {schema, idSchema, name} = this.props;
+    const title = schema.title || name;
+    const {items} = this.state;
+    const {definitions} = this.props.registry;
+    const itemsSchema = retrieveSchema(schema.items, definitions);
+    return (
+      <SelectWidget
           id={idSchema && idSchema.id}
           multiple
           onChange={this.onSelectChange}
           options={optionsList(itemsSchema)}
           schema={schema}
           title={title}
           value={items}
-        />
-      );
-    }
+      />
+    );
+  }
 
+  renderFixedArray() {
+    const {schema, uiSchema, errorSchema, idSchema, name} = this.props;
+    const title = schema.title || name;
+    const {items} = this.state;
+    const {definitions} = this.props.registry;
+    const itemSchemas = schema.items.map(item => retrieveSchema(item, definitions));
+    const additionalSchema = allowAdditionalItems(schema) ? retrieveSchema(schema.additionalItems, definitions) : null;
+    const additionalItems = additionalSchema && items && (items.length > itemSchemas.length) && items.slice(itemSchemas.length);
     return (
       <fieldset
-        className={`field field-array field-array-of-${itemsSchema.type}`}>
+          className="field field-array field-array-fixed-items">
         {title ? <legend>{title}</legend> : null}
         {schema.description ?
-          <div className="field-description">{schema.description}</div> : null}
+            <div className="field-description">{schema.description}</div> : null}
         <div className="row array-item-list">{
-          items.map((item, index) => {
+          itemSchemas.map((itemSchema, index) => {
             const itemErrorSchema = errorSchema ? errorSchema[index] : undefined;
             const itemIdPrefix = idSchema.id + "_" + index;
-            const itemIdSchema = toIdSchema(itemsSchema, itemIdPrefix, definitions);
-            return (
-              <div key={index}>
-                <div className="col-xs-10">
-                  <SchemaField
-                    schema={itemsSchema}
-                    uiSchema={uiSchema.items}
-                    formData={items[index]}
-                    errorSchema={itemErrorSchema}
-                    idSchema={itemIdSchema}
-                    required={this.isItemRequired(itemsSchema)}
-                    onChange={this.onChangeForIndex(index)}
-                    registry={this.props.registry}/>
-                </div>
-                <div className="col-xs-2 array-item-remove text-right">
-                  <button type="button" className="btn btn-danger col-xs-12"
-                    tabIndex="-1"
-                    onClick={this.onDropIndexClick(index)}>Delete</button>
-                </div>
-              </div>
-            );
+            const itemIdSchema = toIdSchema(itemSchema, itemIdPrefix, definitions);
+            const itemUiSchema = (uiSchema.items instanceof Array) ? uiSchema.items[index] : uiSchema.items || {};
```

Comment:
Side note regarding uiSchema handling; were you able to define a custom widget for additional items? I didn't. This is probably something we'll want to address, either in this patch or as a followup.